force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,9 +169,36 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        //Find Account
+        Id accountId;
+        List<Account> accList = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+                if(accList.isEmpty()){
+                    Account newAcc = new Account();
+                    newAcc.Name = accountName;
+                    insert newAcc;
+                    accountId = newAcc.Id;
+                }
+
+            
+            
+        System.debug(accountId);
+
+        List<Opportunity> oppList = new List<Opportunity>();
         // Loop through each Opportunity name
+        for(String oppName : oppNames){
             // Create an Opportunity with the given name
+            Opportunity newOpp = new Opportunity();
+            newOpp.Name = oppName;
+            newOpp.AccountId = accountId;
+            newOpp.StageName = 'New';
+            newOpp.CloseDate = Date.today().addMonths(3);
+            System.debug(newOpp);
```

Comment:
Is there a way to check if these opportunities exist before we create them?

Reply:
This looks good.